### DISCUSSION TITLE
Specifying an input file from a different directory within CLI's 'opa test'

### DISCUSSION THREAD

**Discussion by stevend-15** [open-policy-agent/community#211] [OPA and Rego]
I am new to opa/rego and have an opa test that I would like to run. Within that test .rego file, I would like to use an input.json file from a different directory. Is there a way to specify that file within a "with input as _____" statement within the test file?  i.e. 

```
test_allow {
    allow with input as <path-to-file>
}

```

My thoughts so far have lead me to trying the -b option but the directories are pretty far apart and I do not want a bundle that large and dependent. Additionally, I have thought about the import statements but I do not have the "-i" option within the opa test subcommand. I've also tried specifying each file (the .rego policy file, the .rego test file, and an input file) within the opa test subcommand to no avail. 

Any help is greatly appreciated

**Comment by anderseknert** [comment]
Hi Steven! The `opa test` command does not support reading files as it is first and foremost meant to test policy rather than data. You could embed that data in a .rego file, and have it included in the test, like:

```rego
package test.fixtures

test_object := {
    "attributes": "here",
    "more": "here",
    "...", "..."
}
```
```rego
package test

import data.test.fixtures

test_something {
    allow with input as fixtures.test_object
}
```

If you are looking at testing large sets of data, perhaps [conftest](https://github.com/open-policy-agent/conftest) would be a good option?

**Reply by stevend-15** [reply]
Thank you, I could certainly do that. However, I would like to be able to abstractly and dynamically test my .json configuration file. The configuration file may change and rather than having to copy that config file to my testing directory every time it changes, I would just like to access it within my repo to ensure that the test still works. 

I've taken a good look at conftest and I'm hypothesizing how it could fit into my use case. Could I use conftest to run a .rego test file? In the conftest docs I've seen them running conftest against a .rego policy file with rules named 'deny', 'warn', and 'violation'.  Will conftest work with rules that do not start with those names? Maybe switching to a conftest testing paradigm will be more useful if I want these dynamic file checks.

**Reply by anderseknert** [reply]
Yeah, you can run "normal" OPA tests with conftest as well, but I think it works best when adopting their way of doing things for testing configuration data. But sometimes you might want a bit of both.
